pkg/csplugin/broker_test.go

```diff
@@ -276,10 +556,16 @@ func TestBrokerRun(t *testing.T) {
 
 	pb.PluginChannel <- ProfileAlert{ProfileID: uint(0), Alert: &models.Alert{}}
 	pb.PluginChannel <- ProfileAlert{ProfileID: uint(0), Alert: &models.Alert{}}
-	time.Sleep(time.Second * 4)
+	time.Sleep(time.Millisecond * 200)
 
-	assert.FileExists(t, "./out")
-	assert.Equal(t, types.GetLineCountForFile("./out"), 2)
+	content, err := ioutil.ReadFile("./out")
+	if err != nil {
+		log.Errorf("Error reading file: %s", err)
+	}
+	var alerts []models.Alert
+	err = json.Unmarshal(content, &alerts)
+	assert.NoError(t, err)
+	assert.Equal(t, 2, len(alerts))
```

Comment:
Just thinking out loud. This means Threshold=1 is not honored and the watcher went with ticker event. Had it followed threshold = 1 then plugin would've fired 2 separate events and `assert.Equal(t, types.GetLineCountForFile("./out"), 2)` would pass.

Reply:
It's because both arrive in burst, we already have another test checking that one item .. sleep ...  one item leads to two different writes